package typed.entity;

message GuildBattle
{
	message Reward
	{
		optional uint64 player				= 1;	// 帮会成员guid
		optional uint32 bonus_id			= 2;	// 帮会成员奖励bonus_id
		optional uint32 status				= 3;	// 0 无奖励， 1 有奖未领取， 2 有奖已领取
	}
	
	message Group
	{
		optional uint64 guild				= 1;	// 帮会guid
		// 位置 说明
		// 1	2
		// 5	6
		// 9	10
		// 13	14
		// 3	4
		// 7	8
		// 11	12
		// 15	16
		optional uint32 position			= 2;	
		optional uint32 battle_num			= 3;	// 战斗场次（16强 8强 4强 2强 1强）
		optional uint32 status				= 4;	// 状态 0 未开始战斗 1 胜利  2 失败
		repeated Reward reward_info 		= 5;	// 帮会奖励列表
	}
	
	required uint64 guid					= 1;	// 帮会战guid
	repeated uint64 guild					= 2;	// 报名的帮会
	optional uint32 current_battle_num		= 3;	// 当前战斗轮次（16强 8强 4强 2强 1强）
	repeated Group group_info				= 4;	// 战斗组信息
}